src/arrayMethodSort.js

```diff
@@ -3,9 +3,46 @@
 /**
  * Implement method Sort
  */
+
+function compareAsString(a, b) {
+  const valueA = String(a);
+  const valueB = String(b);
+
+  if (valueA > valueB) {
+    return 1;
+  }
+
+  if (valueA < valueB) {
+    return -1;
+  }
+
+  return 0;
+}
+
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(compareFunction = compareAsString) {
+    let count;
+
+    if (compareFunction === undefined) {
+      return this;
+    }
```

Comment:
You could improve error handling as when **compareFunction === undefined** it will use the default **compareAsString**
```suggestion
    if (typeof compareFunction !== 'function') {
      throw new Error('compareFunction is not valid')
    }
```